brigade.js

```diff
@@ -0,0 +1,204 @@
+const { events, Job, Group } = require("brigadier");
+
+// **********************************************
+// Globals
+// **********************************************
+
+const projectName = "porter-terraform";
+const projectOrg = "deislabs";
+
+// **********************************************
+// Event Handlers
+// **********************************************
+
+events.on("check_suite:requested", runSuite);
+events.on("check_suite:rerequested", runSuite);
+events.on("check_run:rerequested", runSuite);
+
+events.on("exec", (e, p) => {
```

Comment:
Can you remind me what this does vs the runSuite function? I noticed that if I just do the exec event, it's not enough and I'm wondering what this exec event does.

Reply:
Certainly!  The `exec` event is the default event that the `brig` cli sends on a `brig run <brigade project>` invocation.  Thus, I like having a handler for this event for easy testing locally on a given brigade project, e.g., `brig run vdice/porter-terraform`.

Meanwhile, the `runSuite` function is invoked when the `check_suite/run:*` events come in via GitHub on appropriate GH events (a push to master, a pull request, etc.).  As the functions that `runSuite` calls require appropriate webhook data (branch name, author data, etc. -- all of the usual data that GH POSTs), it isn't conducive to testing locally.

Does this help answer the question?

Reply:
Yup! Thanks that makes sense.